lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm

```diff
@@ -187,6 +188,28 @@ sub _validate_common_properties ($self) {
     $validation->optional('description');
 }
 
+# TODO: this exceed runtime limit for some reason
+sub _check_keep_logs_and_results ($self, $properties) {
+    if ($self->is_parent) {
+        if ( ($properties->{default_keep_logs_in_days} // 0) > ($properties->{default_keep_results_in_days} // 0)) {
+          $self->render(
+            json => {error => '`default_keep_logs_in_days` must be greater than `default_keep_results_in_days`'},
+            status => 400
+          );
+          return 1;
+        }
```

Comment:
Can this be

```suggestion
    return $self->render(
            json => {error => '`default_keep_logs_in_days` must be greater than `default_keep_results_in_days`'},
            status => 400
          ) if $self->is_parent && ($properties->{default_keep_logs_in_days} // 0) > ($properties->{default_keep_results_in_days} // 0);
```

and accordingly for the else branch

Reply:
I'm not sure if could it work like this, I try but `Mojo::Reactor::Poll: I/O watcher failed: A response has already been rendered at /usr/lib/perl5/vendor_perl/5.40.2/Mojolicious/Controller.pm line 189.`